core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -48,7 +49,13 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
           .getInputMetricsForReadMethod(blockResult.readMethod)
         existingMetrics.incBytesRead(blockResult.bytes)
 
-        val iter = blockResult.data.asInstanceOf[Iterator[T]]
+        val buf = blockResult.data.toArray
```

Comment:
Will this lead to OOM? In `BlockStore` we have something like `unrollSafely` to prevent this, should we follow that?